### ISSUE TITLE
Inference speed not same as claim, how to get rtf 0.025?

### ISSUE THREAD

**Issue by khursani8** [k2-fsa/OmniVoice#7]
Hi thanks for sharing this, I tried finetuning with Malay using 100 hour, so far it able to generate audio correctly and even the voice cloning also correct. But the inference speed are slow. Need to wait for sherpa-onnx?

Thanks

**Comment by r1di** [comment]
lol, are you using a h200? thats what they used in their papers

**Comment by khursani8** [comment]
yep got 4 h200

**Comment by zhu-han** [comment]
Our RTF evaluation setup is detailed in our paper. Specifically:
> We evaluate the inference speed of OmniVoice on an H20 GPU using PyTorch as the inference framework. We use a 3-second audio promp to generate 10-second audio clips, and report the real-time factor (RTF) across different batch sizes
and inference steps.

The detailed results are shown in this fig:

<img width="687" height="221" alt="Image" src="https://github.com/user-attachments/assets/ccf6f7be-8483-4b78-b66c-20ca2ca246aa" />

If you have H20 GPU, you can reproduce our results with the following commands:
CUDA_VISIBLE_DEVICES=0 omnivoice-infer-batch --test_list test_rtf.jsonl --warmup 5 --num_step 16 --batch_size 4 --res_dir results/rtf/

The rest_rtf.jsonl file and the ref_rtf.wav file (audio prompt) are uploaded in the zip file.

[rtf_evaluation.zip](https://github.com/user-attachments/files/26452212/rtf_evaluation.zip)

Note that with different machines, different steps, different batch sizes, and different lengths of audio prompts or generated audio, the RTF will be different.

**Comment by khursani8** [comment]
thanks for sharing the command, i able to replicate it LGTM

```
(omnivoice) ➜  OmniVoice git:(master) ✗ CUDA_VISIBLE_DEVICES=0 uv run python -m omnivoice.cli.infer_batch \
      --model exp/omnivoice_malay_fixed/checkpoint-5000 \
      --test_list test_list.jsonl \
      --warmup 2 \
      --num_step 16 \
      --batch_size 4 \
      --res_dir results/
2026-04-03 19:40:03,307 INFO [infer_batch.py:436] Using cuda (1 device(s)). Spawning 1 worker processes.
2026-04-03 19:40:06,999 INFO [infer_batch.py:480] Running batch inference
2026-04-03 19:40:06,999 INFO [infer_batch.py:344] Split 4 samples into 1 batches (fixed batch_size=4, sorted by duration)
Processing samples:   0%|                                                                                                                   | 0/1 [00:00<?, ?it/s]2026-04-03 19:40:10,719 INFO [infer_batch.py:229] [Worker 4094610] Initializing worker on device: cuda:0
Loading weights: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████| 313/313 [00:00<00:00, 1070.96it/s]
Loading weights: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████| 527/527 [00:00<00:00, 1146.79it/s]
2026-04-03 19:40:15,412 INFO [infer_batch.py:238] [Worker 4094610] Running 2 warmup iterations on cuda:0
2026-04-03 19:40:17,466 INFO [infer_batch.py:250] [Worker 4094610] Warmup complete on cuda:0
2026-04-03 19:40:17,466 INFO [infer_batch.py:252] [Worker 4094610] Worker on cuda:0 initialized successfully.
Processing samples: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:11<00:00, 11.15s/it]
2026-04-03 19:40:19,223 INFO [infer_batch.py:529] --- Summary ---
2026-04-03 19:40:19,223 INFO [infer_batch.py:530] Total audio duration: 23.00s
2026-04-03 19:40:19,224 INFO [infer_batch.py:531] Total synthesis time: 0.68s
2026-04-03 19:40:19,224 INFO [infer_batch.py:534] Average RTF: 0.0294
2026-04-03 19:40:19,224 INFO [infer_batch.py:538] Done!
(omnivoice) ➜  OmniVoice git:(master) ✗ 
```
